python/torch_mlir/csrc/base_lazy_backend/mlir_lowering_context.h

```diff
@@ -67,8 +67,12 @@ class TORCH_API TorchMlirLoweringContext : public torch::lazy::LoweringContext {
       const std::vector<int64_t>& param_index,
       bool must_alias = false) override;
 
+  // Check if parameter shape matches result at index.
+  bool CheckResultShape(
+      const BackendDataPtr& parameter_data, int64_t result_idx) override;
+
   // Get the shape of the result tuple component, given by index.
-  c10::optional<torch::lazy::Shape> GetResultShape(size_t index) const override;
+  torch::lazy::Shape GetResultShape(size_t index) const override;
```

Comment:
is `GetResultShape` needed anymore? can it be merged with `CheckResultShape`?

Reply:
I guess it could, let me check to see if there are any other uses of it.